src/order/disjoint.lean

```diff
@@ -320,6 +323,19 @@ calc a ⊓ x ≤ (b ⊔ y) ⊓ x : inf_le_inf hle le_rfl
 lemma le_sup_right_iff_inf_left_le {a b} (h : is_compl x y) : a ≤ b ⊔ y ↔ a ⊓ x ≤ b :=
 ⟨h.inf_left_le_of_le_sup_right, h.symm.dual.inf_left_le_of_le_sup_right⟩
 
+lemma le_inf_sup_inf (h : is_compl x y) : a ≤ b ⊓ x ⊔ c ⊓ y ↔ a ⊓ x ≤ b ∧ a ⊓ y ≤ c :=
+begin
+  refine ⟨λ h', ⟨_, _⟩, λ h', _⟩,
+  { refine (inf_le_inf_right x h').trans _,
+    rw [inf_sup_right,  inf_right_idem, inf_assoc, h.symm.inf_eq_bot, inf_bot_eq, sup_bot_eq],
+    exact inf_le_left },
+  { refine (inf_le_inf_right y h').trans _,
+    rw [inf_sup_right,  inf_right_idem, inf_assoc, h.inf_eq_bot, inf_bot_eq, bot_sup_eq],
+    exact inf_le_left },
+  { rw [← h.codisjoint.inf_left_sup_inf_right a],
+    exact sup_le_sup (le_inf h'.1 inf_le_right) (le_inf h'.2 inf_le_right) }
+end
+
```

Comment:
I'm somewhat suspicious of this lemma, maybe @YaelDillies can tell if its sensible.

Reply:
It does look sensible: You can interpret it as saying that, for a partition `x`, `y`, an element is less than the union of a `x` bit and a `y` bit iff its `x` and `y` bits are respectively less than those.

The proof looks too long, however.

Reply:
@YaelDillies Feel free to golf it (now or after merging this PR).

Reply:
I changed the proof and added another lemma.